### DISCUSSION TITLE
AMS Action Script Server component portability

### DISCUSSION THREAD

**Discussion by ROBERT-MCDOWELL** [zenomt/rtmfp-cpp#23] [General]
I'm currently using AMS with a hundred component I write long time ago and would like to know if the spidermonkey version engin of AMS can be portable to rtmfp-cpp?

**Comment by zenomt** [comment]
i'm not sure what you mean. spidermonkey appears to be written "in C++, Rust, and JavaScript". i expect that this implementation of RTMFP could be adapted into spidermonkey.

Adobe Media Server already includes RTMFP (i even wrote that implementation, many years ago). i don't think that Adobe or the current maintainers of AMS have any interest in changing out the RTMFP implementation. last time i looked (which was not that recently), AMS is not open source.

if you mean "can some open source clone of AMS that's using spidermonkey use this library for RTMFP?", then i expect the answer is "yes, if someone were to do the work".

if you mean "can spidermonkey be integrated into this repository to provide an open source clone of AMS", the answer is "no, that is beyond the scope and purpose of this library and repository".

**Comment by ROBERT-MCDOWELL** [comment]
In fact I just wanted to know if script written in ASS put ins scriptlib/componennts can be interpreted by your server? so I can swtich from AMS to your server....

**Reply by zenomt** [reply]
tcserver has no script interpreter. it is an example of using the RTMFP library that also happens to be a reasonable live media server.

**Reply by ROBERT-MCDOWELL** [reply]
ok understood

**Comment by zenomt** [comment]
closing: after clarification, the functionality asked about is not provided by this project and is beyond its scope.
